test/lit/passes/dae-gc.wast

```diff
@@ -195,3 +195,116 @@
   (nop)
  )
 )
+
+;; Function results can be refined, even if exported.
+(module
+ ;; CHECK:      (type $func (sub (func (param anyref) (result anyref))))
+ (type $func (sub (func (param anyref) (result anyref))))
+
+ ;; CHECK:      (func $export (type $1) (param $x anyref) (result (ref any))
```

Comment:
It would be good to use --all-items in the auto update script so we can see the definition of type $1.